### DISCUSSION TITLE
List of RankTwoTensors for AuxKernal Calculations

### DISCUSSION THREAD

**Discussion by shrituntunroy** [idaholab/moose#28194] [Q&A Modules: Solid mechanics] [answered]
Hi MOOSE Team,

I am a new MOOSE user and am currently learning to use MOOSE.

I have a question regarding the RankTwoAux auxkernel calculations. According to the documentation (Ref: https://mooseframework.inl.gov/source/auxkernels/RankTwoAux.html), the valid symbols for Rank Two Aux are given below:

```bash
[./stress_xy]
  type = RankTwoAux
  rank_two_tensor = stress
  variable = stress_xy
  index_i = 0
  index_j = 1
[../]
```

My question is: where can I find the valid names for different rank two tensors, such as plastic strain and total strain, for these auxkernel calculations? Is there a link or resource that provides a full list?

Thank you for your help.

**Comment by GiudGiud** [comment, marked as answer]
Hello

Use Debug/show_material_props=true to show all material properties, and then you should know which ones are tensors

Guillaume
